### ISSUE TITLE
LdapSessionFactoryTests#testSslTrustIsReloaded failures

### ISSUE THREAD

**Issue by jkakavas** [elastic/elasticsearch#68995]
This fails in 7.x, 7.11, 6.8 

**Build scan**:

https://gradle-enterprise.elastic.co/s/fdaeh2nciqjwg/console-log?task=:x-pack:plugin:security:unitTest
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-2/695/testReport/junit/org.elasticsearch.xpack.security.authc.ldap/LdapSessionFactoryTests/testSslTrustIsReloaded/

**Repro line**:
```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.authc.ldap.LdapSessionFactoryTests.testSslTrustIsReloaded" -Dtests.seed=39B8874359AE7DD3 -Dtests.security.manager=true -Dtests.locale=de-AT -Dtests.timezone=Canada/Pacific -Druntime.java=8
```
**Reproduces locally?**:
No
**Applicable branches**:
7.x, 7.11, 6.8 
**Failure history**:
It started failing upon attempting to backport https://github.com/elastic/elasticsearch/pull/68985
**Failure excerpt**:
Expected exception is not thrown. 
```
junit.framework.AssertionFailedError: Expected exception UncategorizedExecutionException but no exception was thrown
```

**Comment by mark-vieira** [comment]
This is still failing on the 6.8 branch.

https://gradle-enterprise.elastic.co/s/5ar3y7afb2o2s/console-log?task=:x-pack:plugin:security:unitTest

**Comment by ywangd** [comment]
> This is still failing on the 6.8 branch.
> 
> https://gradle-enterprise.elastic.co/s/5ar3y7afb2o2s/console-log?task=:x-pack:plugin:security:unitTest

These are different failures fall into two categories. 

The first is due to TLS 1.0 and 1.1 gets disabled in latest JDK releases, which is the same issue as #72877. I muted the fail tests. 
  - org.elasticsearch.xpack.security.authc.ldap.LdapSessionFactoryTests.testDeprecationWarningIfTls1IsUsed
  - org.elasticsearch.xpack.security.transport.ssl.SslIntegrationTests.testDeprecationWarningWhenHttpClientConnectsWithTLS1
  - org.elasticsearch.xpack.security.transport.ssl.SslIntegrationTests.testServerLogsDeprecationWarningWhenTransportClientConnectsWithTLS1

The second is the same as #72359. Albert muted a few with #72529 (different tests but same error message), but apparently there are more.
  - org.elasticsearch.xpack.security.FIPS140SecureSettingsBootstrapCheckTests.testLegacySecureSettingsIsNotAllowed
  - org.elasticsearch.xpack.security.authc.saml.SamlMetadataCommandTests.testDefaultOptionsWithSigningAndMultipleEncryptionKeys
  - org.elasticsearch.xpack.security.authc.saml.SamlRealmTests.testCreateEncryptionCredentialFromKeyStore
  - org.elasticsearch.xpack.security.authc.saml.SamlRealmTests.testCreateSigningCredentialFromKeyStoreSuccessScenarios
  - org.elasticsearch.xpack.security.authc.saml.SamlRealmTests.testReadIdpMetadataFromHttps
  - org.elasticsearch.xpack.security.authc.saml.SamlRealmTests.testCreateSigningCredentialFromKeyStoreFailureScenarios

But somehow they do not fail for 7.x. Did we apply some gradle magic for 7.x branches? @mark-vieira

**Comment by williamrandolph** [comment]
The failing 6.8 `SamlRealmTests` were muted here: https://github.com/elastic/elasticsearch/issues/73314

I ran into the others related to https://github.com/elastic/elasticsearch/issues/72359 today, so I'm going to go ahead and mute them. Do we need a new issue that covers this wider breadth of tests?

**Comment by jkakavas** [comment]
The remaining of the failures are 6.8 specific only and this has been muted in 6.8. I'm going to close this as won't fix since 6.8 is EOL
